samples/auth-user-sample/authUserFlow.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const {OAuth2Client} = require('google-auth-library');
+const readline = require('readline');
+
+function startRl() {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  rl.on('close', function() {
+    process.exit(0);
+  });
+
+  return rl;
+}
+
+// Download your OAuth2 configuration from the Google
+const keys = require('./oauth2.keys.json');
+
+/**
+ * Create a new OAuth2Client, and go through the OAuth2 content
+ * workflow. Return the full client to the callback.
+ */
+async function getRedirectUrl() {
+  const rl = main.startRl();
+  // Create an oAuth client to authorize the API call.  Secrets are kept in a `keys.json` file,
+  // which should be downloaded from the Google Developers Console.
+  const oAuth2Client = new OAuth2Client(
+    keys.installed.client_id,
+    keys.installed.client_secret,
+    keys.installed.redirect_uris[0]
+  );
+
+  // Generate the url that will be used for the consent dialog.
+  const authorizeUrl = oAuth2Client.generateAuthUrl({
+    access_type: 'offline',
+    scope: 'https://www.googleapis.com/auth/bigquery',
+    prompt: 'consent',
+  });
+
+  console.info(
+    `Please visit this URL to authorize this application: ${authorizeUrl}`
+  );
+  rl.question('Enter the authorization code: ', function(code) {
+    main.exchangeCode(code).then(function() {
+      rl.close();
```

Comment:
Once we exchange the code for a refresh token, we can use it below in "authFlow"

Reply:
This was helpful and I refactored to do it this way.... Then I got other feedback and changed it again....